neptun/src/noise/timers.rs

```diff
@@ -207,6 +230,21 @@ impl Tunn {
         let now = time.duration_since(self.timers.time_started).into();
         self.timers[TimeCurrent] = now;
 
+        // Check which timers to update, and update them
+        let timer_mask = self
+            .timers
+            .timers_to_update_mask
+            .load(std::sync::atomic::Ordering::Relaxed);
+        for timer_name in TimerName::VALUES {
+            if (timer_mask & (1 << (timer_name as u16))) != 0 {
+                self.timer_tick(timer_name);
+            }
+        }
+        // Reset all marked bits
+        self.timers
+            .timers_to_update_mask
+            .store(0, std::sync::atomic::Ordering::Relaxed);
```

Comment:
I think this will miss `mark_timer_to_update` calls that happen during the for loop. I think you want to call `let timer_mask = self.timers.timers_to_update_mask.swap(0, Relaxed)` in the beginning.

If I'm correct, please also add a test that could catch this issue.

Reply:
Changed and added test

Reply:
I don't understand how this test, verifies the fix for the above issue. The issue happens only when the for loop (now in `tick_marked_timers`) runs in parallel with `mark_timer_to_update`. Also, reverting the `swap` change from implementation, doesn't break the test - please add a test that fails with the old version of the code, but passes with the new one.

Reply:
Yes, but because of a bit difficulty for running them in parallel to make sure the issue happens, I decided to mark the timer after reading the bit mask. That timer should be updated in the next cycle of update timers, but in previous version that update was being missed.